File: company/src/main/java/ru/scarlet/company/configs/OtlpConfiguration.java
package ru.scarlet.company.configs;

import io.opentelemetry.exporter.otlp.http.trace.OtlpHttpSpanExporter;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;

@Configuration
public class OtlpConfiguration {

    OtlpHttpSpanExporter otlpHttpSpanExporter(@Value("${tracing.url}") String url){
        return OtlpHttpSpanExporter.builder().setEndpoint(url).build();
    }
}

File: file-service/src/main/java/ru/scarlet/fileservice/configs/OtlpConfiguration.kt
package ru.scarlet.fileservice.configs

import io.opentelemetry.exporter.otlp.http.trace.OtlpHttpSpanExporter
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
open class OtlpConfiguration {

    @Bean
    open fun otlpHttpSpanExporter(@Value("\${tracing.url}") url: String?): OtlpHttpSpanExporter {
        return OtlpHttpSpanExporter.builder().setEndpoint(url!!).build()
    }
}